File: bank-service/src/main/java/rs/edu/raf/model/accounts/BankAccount.java
package rs.edu.raf.model.accounts;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import rs.edu.raf.model.Currency;

import java.math.BigDecimal;

@Entity
@Inheritance(strategy = InheritanceType.JOINED)
@Data
@AllArgsConstructor
@NoArgsConstructor
public abstract class BankAccount {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @Column(unique = true)
    private Long accountNumber;
    private Long owner;
    private BigDecimal balance;
    private BigDecimal availableBalance;
    private Long creator; // employee who created account
    private Long creationDate;
    @ManyToOne
    private Currency currency;
    private Boolean active;
}

File: bank-service-kotlin/src/main/kotlin/org/example/bankservicekotlin/model/accounts/BankAccount.kt
package org.example.bankservicekotlin.model.accounts

import jakarta.persistence.*
import org.example.bankservicekotlin.model.Currency
import java.math.BigDecimal

@Entity
@Inheritance(strategy = InheritanceType.JOINED)
open class BankAccount(
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long?,
    @Column(unique = true)
    var accountNumber: Long?,
    var owner: Long?,
    var balance: BigDecimal,
    var availableBalance: BigDecimal,
    val creator: Long, // employee who created account
    val creationDate: Long,
    @ManyToOne
    val currency: Currency,
    var active: Boolean
) {
}